### ISSUE TITLE
Order gene list on similarity to entered term (not alphabetically)

### ISSUE THREAD

**Issue by AlistairNWard** [iobio/gene.iobio#1074]
Search for the gene F5 and there are so many genes returned that contain F5, that the dropdown is cutoff well before genes beginning with "F" are displayed. This means it's impossible to search for F5 and select it.

Instead of ordering genes in the search dropdown alphabetically, they should be ordered according to their similarity to the entered term. As an exact match, F5 (or genes with a synonym of exactly F5) would be then be the first in list when F5 is entered.

**Comment by tonydisera** [comment]
Agreed. I have reworked the gene search typeahead functionality in release 4.11.  This new behavior will try to match on the gene symbol first. If there are no matches, it will search the gene aliases. In your example, the current behavior results in this long list of 'hits':
<img width="523" alt="Screenshot 2024-06-17 at 5 19 00 PM" src="https://github.com/iobio/gene.iobio/assets/8420103/3e55e2d1-8d40-4b83-a37c-e0236876d61b">

In gene.iobio 4.11, the search on F5 will return only that gene:
<img width="202" alt="Screenshot 2024-06-17 at 5 18 54 PM" src="https://github.com/iobio/gene.iobio/assets/8420103/6ed0a6ca-1f50-44a5-9602-5272d222060d">

**Comment by tonydisera** [comment]
@AlistairNWard, you bring up an interesting point about the order of the gene names returned. In the new release, the search looks for a match based on the beginning of the gene name. _Hopefully, this isn't too restrictive._ If we returned all genes that match the term anywhere in the gene name, then the order of the genes returned is more relevant. And I agree with you. The user would want to see the 'closest' matches first. And the new behavior does satisfy this. For example, if the user searches on gene TAT, that exact match appears first in the list:
<img width="170" alt="Screenshot 2024-06-17 at 5 39 44 PM" src="https://github.com/iobio/gene.iobio/assets/8420103/895143b0-7a7e-4f05-88d9-cf031eef4dc5">

And on a related note, gene list order is dictated by the gene name, not the gene alias. So, for example, if the user enters MGC445, there are not any genes with this name, but there are gene aliases that start with this term. Notice that the genes are ordered al
<img width="234" alt="Screenshot 2024-06-17 at 5 43 45 PM" src="https://github.com/iobio/gene.iobio/assets/8420103/e5a87a6a-1e8e-4dc1-b238-a7e1f2115699">
phabetically by the gene name (the name designated by RefSeq or Gencode).
